lib/ProductOpener/Ecoscore.pm

```diff
@@ -496,6 +685,8 @@ The percentages are stored in $aggregated_origins_ref
 
 =cut
 
+sub aggregate_origins_of_ingredients($$$);
+
```

Comment:
is this intended? i don't know if it's duplication, or some perl feature i don't know about. :)

Reply:
It's intended, the function is recursive, and otherwise Perl emits a warning that the function is "called too early to check prototype". The solution to avoid it is to predeclare the function.

See https://www.perlmonks.org/bare/?node_id=130071

Reply:
fascinating. i have seen that message before in the logs, didn't know that was the/a cause.